fms_extras/utils/cache/__init__.py

```diff
@@ -0,0 +1,245 @@
+import abc
+import dataclasses
+from typing import List, Optional, Tuple
+
+import torch
+
+
+def compute_position_ids(
+    num_tokens_per_sequence: List[int], context_lengths: Optional[List[int]] = None
+) -> List[List[int]]:
+    """Compute position ids based on the current context lengths and the new tokens to add
+
+    Parameters
+    ----------
+    num_tokens_per_sequence: List[int]
+        number of tokens to be added to each sequence
+    context_lengths: List[int], optional
+        optional list of current context lengths per sequence. If none, will assume no context length and starting
```

Comment:
the explanation here seems to describe the length of the context (prompt?) being provided to the model... but the code below seems to suggest that this is the start_idx of each row after padding. I think there are a lot of ways to interpret this so we should try to make it less ambiguous.

Reply:
This has been removed in latest commit